### PR TITLE
WRO-9096: Update react-redux related documentation

### PR DESCRIPTION
**Pull request by vladut-clapou-lgp**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
The redux related documentation was outdated so it needed an update.
 - Added Redux-Toolkit documentation.
 - Deleted the part with the presentational and container components because it's deprecated, as stated in the medium article - 
 https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0#.6j9fz9g5j

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRO-9096

### Comments
Enact-DCO-1.0-Signed-off-by: Clapou Vladut [vladut.clapou@lgepartner.com](mailto:vladut.clapou@lgepartner.com)

### CODE REVIEW

**Review by stanca-pop-lgp — APPROVED** [review]
LGTM

**Comment by adrian-cocoara-lgp** [comment]
Please update changelog

**Review by adrian-cocoara-lgp — APPROVED** [review]
LGTM

**Comment by MikyungKim** [comment]
Please rebase your branch with the latest develop branch.

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,12 @@
 
 The following is a curated list of changes in the Enact project, newest changes on the top.
 
+## [unreleased]
+
+### Changed
+
+- Updated React-Redux related documentation
```
Comment (MikyungKim) [inline CHANGELOG.md:9]: We don't add a changelog for updating docs.

File: docs/redux/redux-intro.md [hunk 2]
```diff
@@ -9,7 +9,8 @@ This document provides a high-level overview of Redux and how it is used.
 
 ### What is Redux?
 
-Redux is a library that allows you to manage application state. It closely follows React's Flux data flow model and works well with React, though it does not require it. State management has become more complicated due to a mixing of mutability and asynchronicity, and redux tries to resolve this issue by make state mutation **predictable**.
+Redux is a library that allows you to manage application state. It helps you write applications that behave consistently, run in different environments (client, server, and native), and are easy to test. On top of that, it provides a great developer experience, such as [live code editing combined with a time traveling debugger](https://github.com/reduxjs/redux-devtools).
```
Comment (MikyungKim) [inline docs/redux/redux-intro.md:12]: Let's not change this paragraph.

File: docs/redux/redux-intro.md [hunk 3]
```diff
@@ -19,14 +20,92 @@ We recommend using the most recent version of Redux. To see which version of Red
 
 The entire **[state](https://redux.js.org/understanding/thinking-in-redux/glossary#state)** of the application will be represented by one JavaScript object, a **[store](https://redux.js.org/understanding/thinking-in-redux/glossary#store)**.
 
+This makes it easy to create universal apps, as the state from your server can be serialized and hydrated into the client with no extra coding effort. A single state tree also makes it easier to debug or inspect an application; it also enables you to persist your app's state in development, for a faster development cycle. Some functionality which has been traditionally difficult to implement - Undo/Redo, for example - can suddenly become trivial to implement, if all of your state is stored in a single tree.
```
Comment (MikyungKim) [inline docs/redux/redux-intro.md:23]: I think the previous version of the explanation is enough for the `Three Principles of Redux`.
If you want to explain more, how about adding below at the end?
```
Please find more information [here](https://redux.js.org/understanding/thinking-in-redux/three-principles).

File: docs/redux/redux-intro.md [hunk 4]
```diff
@@ -163,26 +242,23 @@ function counter (state = 0, action) {
 			return state;
 	}
 }
```
Comment (MikyungKim) [inline docs/redux/redux-intro.md:244]: Let's add one empty line after 140 line.

Comment (MikyungKim) [reply docs/redux/redux-intro.md:244]: Let's add `;` to https://github.com/enactjs/enact/pull/3081/files#diff-8a5a91a87d99a04436c951e212abd293a95e1164030ac87952439038334d26e5R178.

File: docs/redux/redux-intro.md [hunk 5]
```diff
@@ -163,26 +242,23 @@ function counter (state = 0, action) {
...
-			<p>
-				Clicked: {value} times
-				{' '}
-				<button onClick={onIncrement}>
-					+
-				</button>
-			</p>
-		);
-	}
+const Counter = ({value, onIncrement}) =>  {
```
Comment (MikyungKim) [inline docs/redux/redux-intro.md:245]: Please remove
```
import {Component} from 'react';
```

File: docs/redux/redux-intro.md [hunk 6]
```diff
@@ -163,26 +242,23 @@ function counter (state = 0, action) {
...
+	);
 }
 Counter.propTypes = {
 	onIncrement: PropTypes.func.isRequired,
 	value: PropTypes.number.isRequired
 };
 const store = createStore(counter);
-function render () {
+const render = () => {
 	ReactDOM.render(
```
Comment (MikyungKim) [inline docs/redux/redux-intro.md:262]: Let's update `render` API to `createRoot` from `react-dom/client`.

File: docs/redux/redux-intro.md [hunk 7]
```diff
@@ -195,87 +271,172 @@ render();
 store.subscribe(render);
 ```
 
-Live Demo: [http://jsbin.com/nemofa/edit?html,js,output](http://jsbin.com/nemofa/edit?html,js,output)
+Live Demo: [https://jsbin.com/pudurig/edit?html,js,output](https://jsbin.com/pudurig/edit?html,js,output)
+
+## Redux Toolkit
```
Comment (MikyungKim) [inline docs/redux/redux-intro.md:276]: I guess you meant ### instead of ##?

File: docs/redux/redux-intro.md [hunk 8]
```diff
@@ -195,87 +271,172 @@ render();
...
+
+### What is Redux Toolkit?
```
Comment (MikyungKim) [inline docs/redux/redux-intro.md:278]: I guess this line is redundant.

File: docs/redux/redux-intro.md [hunk 9]
```diff
@@ -10,6 +10,7 @@ This document provides a high-level overview of Redux and how it is used.
 ### What is Redux?
 
 Redux is a library that allows you to manage application state. It closely follows React's Flux data flow model and works well with React, though it does not require it. State management has become more complicated due to a mixing of mutability and asynchronicity, and redux tries to resolve this issue by make state mutation **predictable**.
+State management has become more complicated due to a mixing of mutability and asynchronicity, and redux tries to resolve this issue by make state mutation **predictable**.
```
Comment (MikyungKim) [inline docs/redux/redux-intro.md:13]: Please remove this change.

File: docs/redux/redux-intro.md [hunk 10]
```diff
@@ -152,8 +157,7 @@ Live demo: [http://jsbin.com/keyahus/edit?html,js,output](http://jsbin.com/keyah
 ```js
 import {createStore} from 'redux';
 import PropTypes from 'prop-types';
-import {Component} from 'react';
-import ReactDOM from 'react-dom';
+import { createRoot } from 'react-dom/client';
```
Comment (MikyungKim) [inline docs/redux/redux-intro.md:160]:
```suggestion
import {createRoot} from 'react-dom/client';
```

File: docs/redux/redux-intro.md [hunk 11]
```diff
@@ -163,119 +167,200 @@ function counter (state = 0, action) {
...
+  const appElement = (
+    <Counter
 			value={store.getState()}
 			onIncrement={() => store.dispatch({type: 'INCREMENT'})}
-		/>,
-		document.getElementById('root')
-	);
+		/>
+  );
+  createRoot(document.getElementById('root')).render(appElement);
```
Comment (MikyungKim) [inline docs/redux/redux-intro.md:193]: Please check the indentation.

File: docs/redux/redux-intro.md [hunk 12]
```diff
@@ -163,119 +167,200 @@ function counter (state = 0, action) {
...
 }
 render();
 store.subscribe(render);
 ```
 
-Live Demo: [http://jsbin.com/nemofa/edit?html,js,output](http://jsbin.com/nemofa/edit?html,js,output)
+Live Demo: [https://jsbin.com/pudurig/edit?html,js,output](https://jsbin.com/pudurig/edit?html,js,output)
```
Comment (MikyungKim) [inline docs/redux/redux-intro.md:199]: Please update the above change to this url.

File: docs/redux/redux-intro.md [hunk 13]
```diff
@@ -163,59 +167,70 @@ function counter (state = 0, action) {
...
 
-### Redux and React
+### Redux Toolkit
 
-As mentioned above Redux can be used without React. React bindings for redux is available from [react-redux](https://github.com/reduxjs/react-redux), which is a generic library that connects React components to a Redux store. More on how to use it is available [here](https://redux.js.org/tutorials/fundamentals/part-5-ui-react).
+The patterns shown above, unfortunately, require lots of verbose and repetitive code. To make it easier to write Redux applications in general, Redux team introduced [Redux Toolkit](https://redux-toolkit.js.org/).
```
Comment (MikyungKim) [inline docs/redux/redux-intro.md:203]: I've updated the rest of the part. Please take a look.

File: docs/redux/redux-intro.md [hunk 14]
```diff
@@ -163,59 +167,70 @@ function counter (state = 0, action) {
...
+Also, we use [useDispatch()](https://github.com/reduxjs/react-redux/blob/master/docs/api/hooks.md#usedispatch) hook to dispatch actions. It gives you the store's `dispatch` method as its result so that you can call it with some `action` to dispatch.
 
-#### Example
+Our components need access to the Redux store so they can subscribe to it. This can be cumbersome as your number of components grows and you have to manually pass store around. `react-redux` incorporates [context](https://reactjs.org/docs/context.html) in React and provides a [`<Provider />`](https://github.com/reduxjs/react-redux/blob/master/docs/api/Provider.md) component to make store available to all components without passings stores around by hand. You only need to use it once at the `render()` of root component.
+
+#### example
 
 ```js
 import {createStore} from 'redux';
```
Comment (MikyungKim) [inline docs/redux/redux-intro.md:236]: Would you please update the current sample to use `useSelector` and `useDispatch`? Thanks.

File: docs/redux/redux-async.md [hunk 15]
```diff
@@ -115,17 +115,16 @@ import {Component} from 'react';
 import {connect} from 'react-redux';
```
Comment (MikyungKim) [inline docs/redux/redux-async.md:115]: The `LS2Request Example` seems not to use `redux-thunk` and still uses `connect`.
Would you please update the example to use `redux-thunk` and `useDispatch`?
Also, I found our pattern-ls2request sample (https://github.com/enactjs/samples/tree/master/ui/pattern-ls2request/src) seems old style. Would you please update it as we did in sandstone/pattern-virtuallist-preserving-focus sample (https://github.com/enactjs/samples/tree/master/sandstone/pattern-virtuallist-preserving-focus/src )?

File: docs/redux/redux-intro.md (hunk 12 above)
Comment (vladut-clapou-lgp) [reply docs/redux/redux-intro.md:199]: Which one?

Comment (MikyungKim) [reply docs/redux/redux-intro.md:199]: I meant, I can't see `createRoot` pattern in the Live Demo. https://jsbin.com/pudurig/edit?html,js,output

Comment (vladut-clapou-lgp) [reply docs/redux/redux-intro.md:199]: Ah, all right, got it!

File: docs/redux/redux-async.md (hunk 15 above)
Comment (vladut-clapou-lgp) [reply docs/redux/redux-async.md:115]: We use `applyMiddleware` from `redux-thunk` to create a thunk action. I will update the sample to use `useDispatch` .

Comment (MikyungKim) [reply docs/redux/redux-async.md:115]: Oh.. I miss worded. not `redux-thunk` but `redux-toolkit. sorry.

File: docs/redux/redux-intro.md (hunk 12 above)
Comment (vladut-clapou-lgp) [reply docs/redux/redux-intro.md]: I'm using stackblitz.com now instead of jsbin because I had some import issues with jsbin while changing the Live Demo sample

Comment (MikyungKim) [reply docs/redux/redux-intro.md]: How about using code sandbox(https://codesandbox.io)?

Comment (vladut-clapou-lgp) [reply docs/redux/redux-intro.md]: That's even better 👍 I changed it to codesandbox, thanks for the tip!

File: docs/redux/redux-async.md [hunk 16]
```diff
@@ -98,24 +93,33 @@ export const getSystemSettings = params => dispatch => {
...
 	}
-}
+});
+
+export const {receiveSystemSettings, updateSystemSettings} = rootSlice.actions;
+export default rootSlice;
 ```
 
-Connected container dispatches ``getSystemSettings`` on componentDidMount and renders a ``pictureMode`` prop that's been hooked up with a redux store.
+Connected container dispatches ``getSystemSettings`` on component mout and renders a ``pictureMode`` prop that's been hooked up with a redux store.
```
Comment (daniel-stoian-lgp) [inline docs/redux/redux-async.md:113]: typo "mout" --> "mount"

File: docs/redux/redux-intro.md [hunk 17]
```diff
@@ -163,111 +167,113 @@ function counter (state = 0, action) {
...
+const root = createRoot(rootElement);
+
+root.render(
+    <Provider store={store}>
+      <App />
+    </Provider>
+);
 ```
 
 Live Demo: [http://jsbin.com/zukojok/1/edit?html,js,output](http://jsbin.com/zukojok/1/edit?html,js,output)
```
Comment (MikyungKim) [inline docs/redux/redux-intro.md:279]: Please update the live demo along with the changed code.

File: docs/redux/redux-intro.md [hunk 18]
```diff
@@ -163,111 +167,113 @@ function counter (state = 0, action) {
...
-}
-const mapStateToProps = (state) => {
-	return {value: state};
 };
-const mapDispatchToProps = (dispatch) => {
-	return {
-		onIncrement: () => dispatch({type: 'INCREMENT'})
+
+//store
+const store = createStore(counterReducer);
```
Comment (MikyungKim) [inline docs/redux/redux-intro.md:247]: Why don't we use `configureStore` from redux-toolkit instead of this?

File: docs/redux/redux-intro.md [hunk 19]
```diff
@@ -163,111 +167,113 @@ function counter (state = 0, action) {
...
 
-Container components need access to the Redux store so they can subscribe to it. This can be cumbersome as your number of components grows and you have to manually pass store around. `react-redux` incorporates [context](https://reactjs.org/docs/context.html) in React and provides a [`<Provider />`](https://github.com/reduxjs/react-redux/blob/master/docs/api/Provider.md) component to make store available to all container components without passings stores around by hand. You only need to use it once at the `render()` of root component.
+Our components need access to the Redux store so they can subscribe to it. This can be cumbersome as your number of components grows and you have to manually pass store around. `react-redux` incorporates [context](https://reactjs.org/docs/context.html) in React and provides a [`<Provider />`](https://github.com/reduxjs/react-redux/blob/master/docs/api/Provider.md) component to make store available to all components without passings stores around by hand. You only need to use it once at the `render()` of root component.
 
-#### Example
+#### example
 
 ```js
+import {createRoot} from 'react-dom/client';
```
Comment (MikyungKim) [inline docs/redux/redux-intro.md:232]: Have you run this code? I'm seeing many eslint warnings from your code.
Please fix them.

File: docs/redux/redux-async.md [hunk 20]
```diff
@@ -55,30 +55,25 @@ render(
...
+
+
+export default function configureAppStore (initialState) {
+	return configureStore({
+		reducer: rootSlice.reducer,
 		initialState,
-		applyMiddleware(thunkMiddleware) // lets us dispatch functions
-	);
-	return store;
+		middleware: [thunkMiddleware]
```
Comment (MikyungKim) [inline docs/redux/redux-async.md:68]: As we read from the redux-toolkit docs, https://redux-toolkit.js.org/introduction/getting-started#whats-included, `redux-thunk` is included by default. Would you please update the documentation based on this information as well as the code?

Comment (vladut-clapou-lgp) [reply docs/redux/redux-async.md:68]: https://redux-toolkit.js.org/api/configureStore#middleware we had `applyMiddleware` in the older code, so I had to provide the middleware we used before using the new syntax, passing it to the `middleware` field in the `configureStore`

File: docs/redux/redux-intro.md [hunk 21]
```diff
@@ -150,132 +154,167 @@ Live demo: [http://jsbin.com/keyahus/edit?html,js,output](http://jsbin.com/keyah
 #### React
 
 ```js
-import {createStore} from 'redux';
-import PropTypes from 'prop-types';
-import {Component} from 'react';
-import ReactDOM from 'react-dom';
+import {createRoot} from 'react-dom/client';
+import {configureStore, createSlice} from '@reduxjs/toolkit';
```
Comment (MikyungKim) [inline docs/redux/redux-intro.md:158]: This example should not use redux toolkit since we didn't introduce it yet.
Please revert the change you did to use redux toolkit here.

File: docs/redux/redux-intro.md [hunk 22]
```diff
@@ -150,132 +154,167 @@ Live demo: [http://jsbin.com/keyahus/edit?html,js,output](http://jsbin.com/keyah
...
+`useSelector()` accepts a single function, which we call a selector function. A selector is a function that takes the entire Redux store state as its argument, reads some value from the state, and returns that result.
 
-#### Example
+Also, we use [useDispatch()](https://github.com/reduxjs/react-redux/blob/master/docs/api/hooks.md#usedispatch) hook to dispatch actions. It gives you the store's `dispatch` method as its result so that you can call it with some `action` to dispatch.
+
+Our components need access to the Redux store so they can subscribe to it. This can be cumbersome as your number of components grows and you have to manually pass store around. `react-redux` incorporates [context](https://reactjs.org/docs/context.html) in React and provides a [`<Provider />`](https://github.com/reduxjs/react-redux/blob/master/docs/api/Provider.md) component to make store available to all components without passings stores around by hand. You only need to use it once at the `render()` of root component.
+
+#### example
```
Comment (MikyungKim) [inline docs/redux/redux-intro.md:250]:
```suggestion
#### Example
```

File: docs/redux/redux-intro.md [hunk 23]
```diff
@@ -150,132 +154,167 @@ Live demo: [http://jsbin.com/keyahus/edit?html,js,output](http://jsbin.com/keyah
...
+
+	const incrementHandler = () => {
+		dispatch(increment());
 	};
+
+	return (
+		<p>
+			Clicked: {value} times
+			<button
+				onClick={// eslint-disable-line react/jsx-no-bind
```
Comment (MikyungKim) [inline docs/redux/redux-intro.md:290]: Can we use `useCallback` instead of this?

**Review by MikyungKim — APPROVED** [review]
:lgtm: Great job @vladut-clapou-lgp!
